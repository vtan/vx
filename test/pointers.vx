(
  (let p 0)
  (let a 1)
  (set p (addr-of a))
  (let b (load64 p))
  (store64 p (+ b 1))
  (set p 0)
  (let c 0x1337)
  (let d (load8 (+ (addr-of c) 1)))
  (store64 (+ (addr-of c) 1) (+ 1 d))

  (
    (let write 1)
    (let stdout 1)
    (let _ (syscall write stdout (addr-of d) (* 4 8)))
  )
)
