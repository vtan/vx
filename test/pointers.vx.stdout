00000000: 1300 0000 0000 0000 3714 0000 0000 0000  ........7.......
00000010: 0000 0000 0000 0000 0200 0000 0000 0000  ................
